pkg/cmd/generate-pg-catalog/main.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// This connects to Postgresql and retrieves all the information about
+// the tables in the pg_catalog schema. The result is printed into a
+// comma separated lines which are meant to be store at a CSV and used
+// to test that cockroachdb pg_catalog is up to date.
+//
+// This accepts the following arguments:
+//
+// -user: to change default pg username of `postgres`
+// -addr: to change default pg address of `localhost:5432`
+//
+// Output of this file should generate:
+// pkg/sql/testdata/pg_catalog_tables
+package main
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/jackc/pgx"
+)
+
+// This query uses udt_name::regtype instead of data_type column because
+// data_type only says "ARRAY" but does not say which kind of array it is.
+const getPgCatalogSQL = `
+	SELECT table_name, column_name, udt_name::regtype data_type, udt_name::regtype::oid data_type_oid
+	FROM information_schema.columns 
+	WHERE table_schema = 'pg_catalog'
+	ORDER BY 1, 2;
+`
+
+const getServerVersion = `SELECT current_setting('server_version');`
+
+var (
+	postgresAddr = flag.String("addr", "localhost:5432", "Postgres server address")
+	postgresUser = flag.String("user", "postgres", "Postgres user")
+)
+
+func main() {
+	flag.Parse()
+	db := connect()
+	defer closeDB(db)
+	fmt.Println(getPGVersion(db))
+	rows := describePgCatalog(db)
+	defer rows.Close()
+
+	for rows.Next() {
+		var table, column, dataType string
+		var dataTypeOid uint32
+		if err := rows.Scan(&table, &column, &dataType, &dataTypeOid); err != nil {
+			panic(err)
+		}
+		fmt.Printf("%s,%s,%s,%d\n", table, column, dataType, dataTypeOid)
```

Comment:
i think it could be useful to have the output file include the names of each column in the output. i mean, at the top, add one row that has the column names (table_name, column_name, data_type, data_type_oid).

or actually, maybe a JSON output here would also be useful. up to you, but just make sure to include the names somewhere

Reply:
Done